lineoperations/src/lineoperations.c

```diff
@@ -0,0 +1,193 @@
+*/
+
+
+#include "geanyplugin.h"
+#include "Scintilla.h"
+#include "linefunctions.h"
+
+
+GeanyPlugin		*geany_plugin;
+GeanyData		*geany_data;
+
+PLUGIN_VERSION_CHECK(224)
+
+PLUGIN_SET_INFO(_("LineOperations"),
+				 _("Line operations, remove duplicate lines, empty lines, lines with only whitespace, sort lines."),
+				 "0.1",
+				 "Sylvan Mostert")
+
+
+static GtkWidget *main_menu_item = NULL;
+
+
+// Remove Duplicate Lines, sorted
+static void
+action_rmdupst_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmdupst(doc);
+}
+
+
+// Remove Duplicate Lines, ordered
+static void
+action_rmdupln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmdupln(doc);
+}
+
+
+// Remove Unique Lines
+static void
+action_rmunqln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmunqln(doc);
+}
+
+
+// Remove Empty Lines
+static void
+action_rmemtyln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmemtyln(doc);
+}
+
+
+// Remove Whitespace Lines
+static void
+action_rmwhspln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmwhspln(doc);
+}
+
+
+// Sort Lines Ascending
+static void
+action_sortasc_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) sortlines(doc, 1);
+}
+
+
+// Sort Lines Descending
+static void
+action_sortdesc_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) sortlines(doc, 0);
+}
+
+
+void plugin_init(GeanyData *data)
+{
+	GtkWidget *submenu;
+	GtkWidget *sep1;
+	GtkWidget *sep2;
+	GtkWidget *rmdupst_item;
+	GtkWidget *rmdupln_item;
+	GtkWidget *rmunqln_item;
+	GtkWidget *rmemtyln_item;
+	GtkWidget *rmwhspln_item;
+	GtkWidget *sortasc_item;
+	GtkWidget *sortdesc_item;
+
+	/* Add an item to the Tools menu */
+	main_menu_item = gtk_menu_item_new_with_mnemonic(_("Line Operations"));
+	gtk_widget_show(main_menu_item);
+	ui_add_document_sensitive(main_menu_item);
+
+	submenu = gtk_menu_new();
+	gtk_widget_show(submenu);
+	sep1 = gtk_separator_menu_item_new();
+	sep2 = gtk_separator_menu_item_new();
+	rmdupst_item  = gtk_menu_item_new_with_mnemonic(_("Remove Duplicate Lines, Sorted"));
+	rmdupln_item  = gtk_menu_item_new_with_mnemonic(_("Remove Duplicate Lines, Ordered"));
+	rmunqln_item  = gtk_menu_item_new_with_mnemonic(_("Remove Unique Lines"));
+	rmemtyln_item = gtk_menu_item_new_with_mnemonic(_("Remove Empty Lines"));
+	rmwhspln_item = gtk_menu_item_new_with_mnemonic(_("Remove Whitespace Lines"));
+	sortasc_item  = gtk_menu_item_new_with_mnemonic(_("Sort Lines Ascending"));
+	sortdesc_item = gtk_menu_item_new_with_mnemonic(_("Sort Lines Descending"));
```

Comment:
Yeah, I guess it's acceptable (and several plugins already do that).  As you said, it can indeed be suboptimal as another plugin might have the same mnemonic as yours, but as @codebrainz said it wouldn't break things too bad either.
So IMO do as you like.  but maybe @frlan has a stronger opinion on that matter?